### DISCUSSION TITLE
LiDAR Packets and Lossy Compression

### DISCUSSION THREAD

**Discussion by shrijitsingh99** [facontidavide/pointcloud_compression#4] [Q&A]
Hello,
Very exciting work, this will be very useful for the community.

I wanted to point out a few things:
1. Lossy Compression
For some LiDAR's like Hesai XT32, they have dual return modes. In this mode, if the LiDAR beam doesn't have a different return it just returns the duplicated point as the first return. So in that sense filtering out those duplicate points can reduce the point cloud by upto 0.5x.
I am not sure if that was the case in the datasets you are testing.

2. LiDAR Packets
It would be great to have a comparison of size with the raw LiDAR packets, I know Hesai LiDARs provide that. They are generally smaller than the PointCloud messages.

3. Smaller PointCloud Messages
Recently a PR was merged which removed padding while converting PointClouds to the msg type https://github.com/ros-perception/perception_pcl/pull/449
Would be an interesting to see the what kind of compression we can get after removing the padding.

**Comment by facontidavide** [comment]
1. No, it is not, but I am sure we can take advantage of that. At the end of the day, I am building a customized coded for each LIDAR manufacturer because... why not ?

2. Raw packet will always win. If you can do that, you should! 

3. That is great! Of course this will help us already. But my method will further improve the size. For instance, most of the rosbag I am testing have NO padding natively, and size is reduced 3X

**Comment by facontidavide** [comment]
would you be so kind to share a rosbag with me?

**Reply by shrijitsingh99** [reply]
Yes, happy to share a few bags from Hesai XT32, Ouster OS1 and RoboSense. Would you prefer small bag files or large ones? I have some really larges ones which are in the 20gb+ range when uncompressed.

**Reply by facontidavide** [reply]
Small, please

**Reply by facontidavide** [reply]
I would really love to get my hands on those rosbags...

**Reply by shrijitsingh99** [reply]
Emailed it to you today

**Reply by alitekes1** [reply]
Can you share with me rosbag files from robosense airy?
